src/main/kotlin/org/rust/lang/core/psi/ext/RsBaseType.kt

```diff
@@ -6,14 +6,43 @@
 package org.rust.lang.core.psi.ext
 
 import org.rust.lang.core.psi.RsBaseType
+import org.rust.lang.core.psi.RsElementTypes.*
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.tokenSetOf
 
-val RsBaseType.isCself: Boolean get() {
-    val path = path
-    return path != null && !path.hasColonColon && path.hasCself
+val RsBaseType.name: String? get() = path?.referenceName
+
+val RsBaseType.kind: RsBaseTypeKind
+    get() = when (kindEnum) {
+        RsBaseTypeKindEnum.UNIT -> RsBaseTypeKind.BtUnit
+        RsBaseTypeKindEnum.NEVER -> RsBaseTypeKind.BtNever
+        RsBaseTypeKindEnum.UNDERSCORE -> RsBaseTypeKind.BtUnderscore
+        RsBaseTypeKindEnum.PATH -> RsBaseTypeKind.BtPath(path ?: error("Malformed RsBaseType element: `$text`"))
+    }
+
+sealed class RsBaseTypeKind {
+    object BtUnit : RsBaseTypeKind()
+    object BtNever : RsBaseTypeKind()
+    object BtUnderscore : RsBaseTypeKind()
+    data class BtPath(val path: RsPath) : RsBaseTypeKind()
```

Comment:
Do we really need `Bt` prefix? Looks like it doesn't add any additional info because it's not obvious that `Bt` means `BaseType`. I'd prefer to use this types like `RsBaseTypeKind.Unit` - it's more informative although longer

Reply:
There is `kotlin.Unit` type... :thinking:

Reply:
That's why I suggest using these classes/objects like `RsBaseTypeKind.Unit` instead of `Unit`